### ISSUE TITLE
What's in a CASE?

### ISSUE THREAD

**Issue by jedwards4b** [CESM-Development/cime#186]
Now that we've completely moved away from the idea that a case directory should be 
independent of a CIME source tree I would like to consider removing some of the other 
files that we copy to the case directory that we don't need to copy.   This list includes:

case_setup
preview_namelist
archive_metadata.sh
check_case - move
check_input_data - move
create_production_test - move
st_archive - move
testcase_setup - move
xmlchange
xmlquery

I propose that these tools be left in their repository location (which could possibly be moved in the repository).

Another, perhaps independent issue, is the scripts which get a case name prefix:
Some of these are created from templates and really are unique to each case but
others are not,  $CASE.submit for example is just a copy of Tools/submit 
So I think that submit and others that are just copies should be in the list above.   

For those that are templates such as $CASE.run
I think that maybe we should move completely away from $CASE prefix and just call it case.run, case.st_archive, etc.

**Comment by billsacks** [comment]
I support the idea of copying less to the case directory. But are you suggesting that, rather than (e.g.) running

```
./xmlchange VAR=VALUE
```

a user would instead run the following:?

```
/path/to/cime/that/corresponds/to/this/case/xmlchange VAR=VALUE
```

If I'm understanding that right, then to me, the only way this would be a workable change is if we had some sort of wrapper script in the case directory that would allow you to run something like:

```
./run xmlchange VAR=VALUE
```

where, under the covers, the 'run' script (which probably needs to be named something else) takes its first argument and prefixes it with the full path to the cime directory in question.

However, even with that, there's still a part of me that feels this is taking the "don't copy things to the case directory" idea too far - unless there are real benefits in not copying these files to the case directory (in terms of maintainability, keeping other scripts simpler, etc.)

I absolutely support renaming the $CASE.run (etc.) scripts to case.run.

**Comment by jedwards4b** [comment]
On Fri, Sep 25, 2015 at 1:01 PM, Bill Sacks notifications@github.com
wrote:

> I support the idea of copying less to the case directory. But are you
> suggesting that, rather than (e.g.) running
> 
> ./xmlchange VAR=VALUE
> 
> a user would instead run the following:?
> 
> /path/to/cime/that/corresponds/to/this/case/xmlchange VAR=VALUE
> 
> ​No, this doesn't seem like a feasible approach , I was thinking of
> something more like adding cime/scripts/Tools to the PATH.
> ​So you would just run xmlchange VAR=VALUE.      ​
> 
>  If I'm understanding that right, then to me, the only way this would be
> a workable change is if we had some sort of wrapper script in the case
> directory that would allow you to run something like:
> 
> ./run xmlchange VAR=VALUE
> 
> where, under the covers, the 'run' script (which probably needs to be
> named something else) takes its first argument and prefixes it with the
> full path to the cime directory in question.
> 
> However, even with that, there's still a part of me that feels this is
> taking the "don't copy things to the case directory" idea too far - unless
> there are real benefits in not copying these files to the case directory
> (in terms of maintainability, keeping other scripts simpler, etc.)
> 
> I absolutely support renaming the $CASE.run (etc.) scripts to case.run.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CESM-Development/cime/issues/186#issuecomment-143323366
> .

## 

Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO

**Comment by mvertens** [comment]
I don't think this is a good idea. I do think this is definitely taking
things to far - and will only lead to confusion for both current and new
users.
But I would prefer to bring this up at the CSEG meeting with a wider
discussion on Monday's meeting.

I do agree that $CASE.run should be case.run and $CASE.build should be
case.build etc. That does seem like a nice simplification.

Mariana

**Comment by rljacob** [comment]
I feel like I'm coming in in the middle of this but my thoughts:  if the file is meant to be altered to run the case (or "power user" variations on the case), copy it to the case directory.  Otherwise, one could leave it where it is.  But requiring users to alter their path as part of installing the model might be a bit much.  And something like $model.run or cime.run would make more sense.

**Comment by jedwards4b** [comment]
Resolved by #231
